volga/streaming/runtime/network/buffer/buffer_memory_tracker.py

```diff
@@ -0,0 +1,78 @@
+import os
+from typing import Optional
+
+import locket
+from shared_memory_dict import SharedMemoryDict
+
+# each input channel has it's own capacity allocation (in buffers, since we have multiple input queues on DataWriter)
+DEFAULT_CAPACITY_PER_IN_CHANNEL = 3 # approx 3 buffers
+
+# for operator's outputs we share capacity per operator (not per channel, since we have one output queue on DataReader)
+DEFAULT_CAPACITY_PER_OUT = 2
+
+
+# TODO add floating buffers
+# multiproc memory allocator for buffers
+# keeps track of used memory per channel
+# uses shared memory to store state + file-based locking
+class BufferMemoryTracker:
+
+    _instances = {}
+
+    DICT_MEMORY_SIZE = 1024
+    LOCKET_LOCK_DIR = '/tmp/volga_locks'
+
+    @staticmethod
+    def instance(
+        node_id: str,
+        capacity_per_in_channel: int = DEFAULT_CAPACITY_PER_IN_CHANNEL,
+        capacity_per_out: int = DEFAULT_CAPACITY_PER_OUT
+    ) -> 'BufferMemoryTracker':
+        if node_id not in BufferMemoryTracker._instances:
+            BufferMemoryTracker._instances[node_id] = BufferMemoryTracker(node_id, capacity_per_in_channel, capacity_per_out)
+        return BufferMemoryTracker._instances[node_id]
+
+    def __init__(self, node_id: str, capacity_per_channel: int, capacity_per_out: int):
+        self._capacity_per_in_channel = capacity_per_channel
+        self._capacity_per_out = capacity_per_out
+        self._node_id = node_id
+        # TODO should we add job_name to dict and lock identifiers?
+        self._shared_dict_name = f'shared-dict-{self._node_id}'
+
+        os.makedirs(BufferMemoryTracker.LOCKET_LOCK_DIR, exist_ok=True)
+        self._lock_name = f'{BufferMemoryTracker.LOCKET_LOCK_DIR}/lock-memory-allocator-{self._node_id}'
```

Comment:
let's use unique folder for tests

Reply:
Passed `job_name` to make lock name unique